src/main/java/de/dafuqs/spectrum/blocks/chests/CompactingChestBlockEntity.java

```diff
@@ -206,4 +224,29 @@ public void writeScreenOpeningData(ServerPlayerEntity player, PacketByteBuf buf)
 		buf.writeInt(this.autoCraftingMode.ordinal());
 	}
 
+	public static void clearCache() {
+		cache.clear();
+	}
+
+	private static record InputItem(Item item, @Nullable NbtCompound nbt) {
+
+		public static InputItem ofItemStack(ItemStack itemStack) {
+			NbtCompound nbt = null;
+			if (itemStack.getNbt() != null)
+				nbt = itemStack.getNbt().copy();
+			return new InputItem(itemStack.getItem(), nbt);
+		}
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) return true;
+			if (o == null || getClass() != o.getClass()) return false;
+			InputItem that = (InputItem) o;
+			return item.equals(that.item) && Objects.equals(nbt, that.nbt);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(item, nbt);
+		}
+	}
```

Comment:
Why not use Fabric API's ItemVariant here?

Reply:
I'm usually modding with forge and didn't knew about the existence of ItemVariant

Reply:
I've updated the PR to use ItemVariant